openfeature/_event_support.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import TYPE_CHECKING, Dict, List
+
+from openfeature.event import (
+    EventDetails,
+    EventHandler,
+    ProviderEvent,
+    ProviderEventDetails,
+)
+from openfeature.provider import FeatureProvider
+
+if TYPE_CHECKING:
+    from openfeature.client import OpenFeatureClient
+
+
+_global_handlers: Dict[ProviderEvent, List[EventHandler]] = defaultdict(list)
+_client_handlers: Dict[
+    OpenFeatureClient, Dict[ProviderEvent, List[EventHandler]]
+] = defaultdict(lambda: defaultdict(list))
+
+
+def run_client_handlers(
+    client: OpenFeatureClient, event: ProviderEvent, details: EventDetails
+) -> None:
+    for handler in _client_handlers[client][event]:
+        handler(details)
+
+
+def run_global_handlers(event: ProviderEvent, details: EventDetails) -> None:
+    for handler in _global_handlers[event]:
+        handler(details)
```

Comment:
depends a bit on the specs, but this will fail with an ugly `KeyError`, if someone invokes it with a non-existing or not yet added event/client. Wouldn't it be better just to log an error message and not through an exception?

Reply:
`_global_handlers` is a `defaultdict(list)` so it should just return an empty list for non-existent events.

Reply:
yeah, you are right 🙈